crates/relayer/src/util/stream.rs

```diff
@@ -90,6 +92,78 @@ where
     }
 }
 
+pub fn try_group_while_timeout<A, E, S, F>(
+    input: S,
+    group_these: F,
+    timeout: Duration,
+) -> impl Stream<Item = Result<Vec<A>, E>>
+where
+    S: Stream<Item = Result<A, E>>,
+    F: Fn(&A, &A) -> bool + 'static,
+{
+    struct State<A> {
+        cur: A,
+        group: Vec<A>,
+    }
+
+    let input = input.timeout(timeout);
+
+    stream! {
+        let mut state: Option<State<A>> = None;
+
+        for await x in input {
+            match x {
+                // timeout
+                Err(e) => {
+                    tracing::warn!("{e}");
```

Comment:
Should we keep this as `warn!`? With smaller delays it might flood the logs

Reply:
The timeout is only emitted at most once per block, but yeah we should downgrade it anyway to debug.